src/utils/helpers.ts

```diff
@@ -33,7 +33,7 @@ export const getLocaleFormat = (dateFormat: string | undefined) => {
  *  A function that takes an array of errors and the component props and throws an error
  *  if the condition is met
  * */
-export const propsHandler = <T>(
+export const errorHandler = <T>(
   errors: { condition: (p: T) => boolean; message: string }[],
   props: T
```

Comment:
Maybe we could add a 3rd optional argument with the exact error we want to have. For example

```js
class PropsValidationError extends Error {
  constructor(message) {
    super(message);
    this.name = "PropsValidationError"; 
  }
}
```

```js
...
errorHandler(validators, props, PropsValidationError);
...
```

Reply:
Great idea on the custom error but in order for this to be cleaner and extendable wouldn't it be better to pass the custom Error instead of the message and just throw it?

Reply:
Unless i am not understandng the question, this is what my example illustrates. You create a custom `Error`, you pass it as a 3rd optional argument, and whatever throws from these validators will be a custom `Error` :thinking:

Reply:
Check the last commit to understand what I meant

Reply:
Ah, I see. My original suggestion was based on the assumption that a given group of validators would fall into a specific "error category" so all exceptions would throw the same kind of error.

Reply:
Yes, but I thought it's better to allow for different errors.

This way, the same component could throw different errors and we could have different handling for each error in the future.    

Also I didn't want to add a third argument in the function to keep it simpler and easier to test.